icu4j/main/core/src/test/java/com/ibm/icu/dev/test/message2/MessageFormat2Test.java

```diff
@@ -574,4 +574,85 @@ public void testVariableOptionsInSelectorWithLocalVar() {
         assertEquals("test local vars loop", "Count = 23, OffCount = 21, and delta=2.",
                 mfVar2.formatToString(Args.of("count", 23, "delta", 2)));
     }
+
+    // Needs more tests. Ported from the equivalent test in ICU4C
+    @Test
+    public void testFormatterAPI() {
```

Comment:
nit (non-blocking, not intended to be acted upon for ICU 76): Ideally, this test would be broken up into 3-4 tests

Reply:
ACK.

This is in sync with the C++ test.
Would need to refactor both.

But I really expect to replace it completely with .json files from the WG.